### PR TITLE
feat: add lstc2022 windows image build

### PR DESCRIPTION
**Pull request by andyzhangx**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
feat: add lstc2022 windows image build
This PR also replaces servercore base image with nanoserver
a servercore base image is 2.6GB, while nanoserver base image is only 100MB

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
feat: add lstc2022 windows image build
```

### CODE REVIEW

**Review by ZeroMagic — APPROVED** [review]
/lgtm

File: hack/release.sh [hunk 1]
```diff
@@ -91,7 +91,7 @@ ALLOW_OVERRIDE=${ALLOW_OVERRIDE:-}
 SKIP_BUILD=${SKIP_BUILD:-}
 SKIP_PUSH_LATEST=${SKIP_PUSH_LATEST:-}
 LINUX_ARCH=${LINUX_ARCH:-amd64 arm arm64 ppc64le s390x}
-WINDOWS_DISTROS=${WINDOWS_DISTROS:-ltsc2019 20H2}
+WINDOWS_DISTROS=${WINDOWS_DISTROS:-1809 20H2 lstc2022}
```
Comment (mauriciopoppe) [inline hack/release.sh:94]: typo, it's ltsc2022, isn't it?

File: Makefile [hunk 2]
```diff
@@ -21,7 +21,7 @@ GOVERSION ?= 1.17
 LINUX_ARCH ?= amd64
 WINDOWS_DISTROS ?=
 
-WINDOWS_BASE_IMAGES=$(addprefix mcr.microsoft.com/windows/servercore:,$(WINDOWS_DISTROS))
+WINDOWS_BASE_IMAGES=$(addprefix mcr.microsoft.com/windows/nanoserver:,$(WINDOWS_DISTROS))
```
Comment (mauriciopoppe) [inline Makefile:24]: I'm not sure if this base image would work, I haven't tested it

Comment (andyzhangx) [reply Makefile:24]: yes, it's fixed, all linux and windows images are built succesfully now: https://console.cloud.google.com/gcr/images/k8s-staging-sig-storage/global/local-volume-provisioner

Comment (andyzhangx) [reply Makefile:24]: btw, we don't want to use `servercore`, that's 2.6GB, it's horrible

File: hack/release.sh (hunk 1 above)
Comment (andyzhangx) [reply hack/release.sh:94]: correct, I will move `WINDOWS_DISTROS` into Makefile in another PR, that would have the test converge in this repo's tests.

**Pull request merged**